nephos/web/info_panel/main/views.py

```diff
@@ -88,23 +91,186 @@ def show_jobs():
+def add_channel():
+    """
+    <url>/add/channel
+
+    View that adds a Channel
+
+    """
+    # Select The Form
+    form = ChannelForm()
+
+    # Validate if the Form statisfies everything
+    if form.validate_on_submit():
+        # Get the Data from the Form
+        name = form.name.data
+        ip = form.ip.data
+        lang = form.lang.data
+        country_code = form.country_code.data
+        timezone = form.timezone.data
+
+        # Insert The Data from the Form into the Database
+        query = DB.session.execute("INSERT INTO channels (name, ip, \
+            country_code, lang, timezone, status) \
+            VALUES ('{}', '{}','{}', '{}', '{}', 'down')"
+                                   .format(name, ip,
+                                           country_code, lang, timezone))
+
+        flash('Channel Added Successfuly!', 'success')
+        return redirect(url_for('main.show_channels'))
+
+    # Instead of Creating an Add template for the same form, i've reused this one
+    return render_template('edit_channel.html', form=form)
+
+
+# ========== Todo: Fix up Database Handlers here ↓
+
+@MAIN_BP.route('/delete/job/<id>', methods=['GET', 'POST'])
+def delete_job(id):
+    """
+    <url>/delete/job/<id>
+
+    View that Renders the Homepage
+
+    """
+
+    # Select The Form
+    form = DeleteForm()
+
+    # Validate if Form Statisfies Everything
+    if form.validate_on_submit():
+        # Execute Deletion but Select the other Database
+        jobs_engine = DB.get_engine(APP, 'jobs')
+        query = jobs_engine.execute(
+            "DELETE FROM apscheduler_jobs WHERE channel_id={}".format(id))
+        flash('Delete Successful!', 'success')
+        return redirect(url_for('main.show_jobs'))
+    return render_template('delete_jobs.html', form=form)
+
+
+@MAIN_BP.route('/edit/job/<id>', methods=['GET', 'POST'])
+def edit_job(id):
+    """
+    <url>/edit/job/<id>
+
+    View that edits a Job
+
+    """
+
+    # Select The Other Database and Find The Record
+    jobs_engine = DB.get_engine(APP, 'jobs')
+    entry = jobs_engine.execute('SELECT * FROM apscheduler_jobs WHERE id={};'
+        .format(id)).fetchone()
+    print(entry)
+    form = JobForm(obj=entry)
+
+    # Validate That Everything is statisfied in the Form
+    if form.validate_on_submit():
+        print(form.data)
+        #query = DB.session.execute("UPDATE apscheduler_jobs SET next_run_time='{}' \
+        #WHERE id={}".format(name, ip, country_code, lang, timezone, id))
+        flash('Edit Successful!', 'success')
+        return redirect(url_for('main.show_jobs'))
+
+    return render_template('edit_job.html', form=form)
+
+
+
+@MAIN_BP.route('/add/job', methods=['GET', 'POST'])
+def add_job():
+    """
+    <url>/add/job
+
+    View that adds a Job
+
+    """
+    form = JobForm()
+
+    if form.validate_on_submit():
+        #DB.session.execute("INSERT INTO apscheduler_jobs (name, ip, \
```

Comment:
This has been left empty; well, on a deeper look I figured you we cannot add jobs from outside the nephos runtime. You'll require some way to connect APScheduler and then mount the job store and then deletem add and edit Jobs.

-https://github.com/viniciuschiele/flask-apscheduler